---
title: " Mass-spec proteomics analysis: Focal adhesion kinase (FAK) in adipose cells (no IP1)"
author:
  name: "Dmytro Kryvokhyzha"
email: dmytro.kryvokhyzha@med.lu.se
affiliation: LUDC Bioinformatics Unit
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: true
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE, warning=FALSE, eval=TRUE)
knitr::opts_knit$set(root.dir = '~/Projects/Karin_Stenkula/proteomic_FAK_adipose/')
```

## Dependencies

```{r, message=FALSE, eval=TRUE}
library(readxl)
library(DEP)
library(dplyr)
library(SummarizedExperiment)
library(ggplot2)
library(ggrepel)
library(assertthat)
```

## Data

### Load

Load the data from the Excel file:

```{r}
data <- read_excel("data/Karin_Stenkula_IP_202312_LH_filtered.xlsx")
```

### Process

Check the structure of the data:

```{r}
str(data)
```

Are there any duplicated accession names?

```{r}
data$v %>% duplicated() %>% any()
```

## Remove IP1 

I remove the sample `IP1` because it is an outlier in a PCA plot
of the previous analysis with all samples. See `results/reports/proteomic_FAK_adipose.html`.

```{r}
data_no_IP1 <- data[,-grep('IP1,', colnames(data))]
data <- data_no_IP1
```

## Experimental design

Create an experimental design table:

```{r}
colnames(data)[c(11:21)] <- gsub(' ', '_', gsub(', ', '_', colnames(data)[c(11:21)]))
labels <- colnames(data)[c(11:21)]

expDesign <- data.frame(label = labels,
                        condition = sub("^[^_]+_", "", labels),
                        replicate = sub("_.+", "", labels)
                        )
# Reverse the order
expDesign <- expDesign %>% arrange(desc(row_number()))

knitr::kable(expDesign)
```

Generate a SummarizedExperiment object:

```{r}
# create the ID and gene name columns:
colnames(data)[2] <- 'ID'
data$name <- sub(".*GN=([^ ]+).*", "\\1", data$Description)

# Generate a SummarizedExperiment object
data_se <- make_se(data, c(11:21), expDesign)
data_se
```

## Filter

Plot a barplot of the protein identification overlap between samples

```{r, fig.width = 4, fig.height = 4}
plot_frequency(data_se)
```

Filter for proteins that are identified in all replicates of at least one condition

```{r}
data_filt <- filter_missval(data_se, thr = 0)
# Less stringent filtering:
# data_filt2 <- filter_missval(data_se, thr = 1)
```

Plot a barplot of the number of identified proteins per samples

```{r, fig.width = 5, fig.height = 6}
plot_numbers(data_filt)
```

Plot a barplot of the protein identification overlap between samples

```{r, fig.width = 4, fig.height = 4}
plot_coverage(data_filt)
```

## Normalize

Normalize the data

```{r}
data_norm <- normalize_vsn(data_filt)
```

Verify the fit:

```{r, fig.width = 5, fig.height = 4}
meanSdPlot(data_norm)
```

Visualize normalization by boxplots for all samples before and after normalization

```{r, fig.width = 6, fig.height = 6}
plot_normalization(data_filt, data_norm)
```

## Impute data for missing values

Plot a heatmap of proteins with missing values

```{r, fig.height = 6, fig.width = 5}
plot_missval(data_filt)
```

Plot intensity distributions and cumulative fraction of proteins with and without missing values

```{r, plot_detect, fig.height = 4, fig.width = 4}
plot_detect(data_filt)
```

The data can be missing at random (MAR), for example if proteins are quantified 
in some replicates but not in others. Data can also be missing not at random (MNAR), 
for example if proteins are not quantified in specific conditions (e.g. in the control samples).

MNAR should be imputed by a left-censored imputation method, such as the quantile 
regression-based left-censored function (“QRILC”) or random draws from a left-shifted 
distribution (“MinProb” and “man”). In contrast, MAR data should be imputed with 
methods such as k-nearest neighbor (“knn”) or maximum likelihood (“MLE”) functions. 

Impute:

```{r, impute, results = "hide", message = FALSE, warning = FALSE, error = TRUE}
data_imp <- impute(data_norm, fun = "knn")
```

Plot intensity distributions before and after imputation

```{r, plot_imp, fig.width = 5, fig.height = 4}
plot_imputation(data_norm, data_imp)
```

## Differential enrichment analysis

Based on linear models and empirical Bayes statistics

```{r, statistics}
data_diff <- test_diff(data_imp, type = "all")
```

Denote significant proteins based on user defined cutoffs

```{r, add_reject}
dep <- add_rejections(data_diff, alpha = 0.05, lfc = 0.5)
```

## Visualization of the results

### PCA

Plot the first and second principal components

```{r, pca, fig.height = 3, fig.width = 6}
plot_pca(dep, x = 1, y = 2, point_size = 4, indicate = "condition")
```

### Correlation matrix

Plot the Pearson correlation matrix

```{r, corr, fig.height = 6, fig.width = 7}
plot_cor(dep, indicate = "condition")
```

### Heatmap

Plot a heatmap of all significant proteins with the data centered per protein

```{r, heatmap, fig.height = 6, fig.width = 7}
plot_heatmap(dep,
             type = "centered",
             kmeans = TRUE,
             show_row_names = FALSE,
             indicate = "condition")
```

Plot a heatmap of all significant proteins (rows) and the tested contrasts (columns)

```{r, heatmap2, fig.height = 8, fig.width = 4}
plot_heatmap(dep, type = "contrast", kmeans = TRUE, 
             k = 6, col_limit = 10, show_row_names = FALSE)
```

### Volcano plot

Plot a volcano plot for the contrast FAK contrasts:

```{r, volcano, fig.height = 5, fig.width = 5}
plot_volcano(dep, contrast = "FAK_DMSO_INS_vs_FAK_DMSO_basal")
plot_volcano(dep, contrast = "FAK_PF_INS_vs_FAK_DMSO_basal")
plot_volcano(dep, contrast = "FAK_PF_INS_vs_FAK_DMSO_INS")
```

### Barplot

Plot a barplot for top 4 proteins from each comparison:

```{r, bar, fig.height = 8, fig.width = 5}
plot_single(dep,
            proteins = c('Ca4', 'Gnai2', 'Acad9', 'Arpc4', 'Mlycd', 'Arpc2'),
            type = "contrast")
```

Plot a barplot for the same protein with the centered log2-intensity data:

```{r, bar2, fig.height = 8, fig.width = 5}
plot_single(dep,
            proteins = c('Ca4', 'Gnai2', 'Acad9', 'Arpc4', 'Mlycd', 'Arpc2'),
            type = "centered")
```

### Frequency plots

Plot a frequency plot of significant proteins for the different contrasts:

```{r, overlap, fig.height = 5, fig.width = 8}
plot_cond(dep)
```
Print the overlap frequency table:

```{r, overlap+_print}
dep_overlap <- plot_cond(dep, plot = F)
knitr::kable(cbind(dep_overlap$counts))
knitr::kable(dep_overlap$legend)
```

## Results table

Generate a results table

```{r, results_table}
data_results <- get_results(dep)
```

Number of significant proteins

```{r, results_table_sign}
data_results %>% filter(significant) %>% nrow()
```

Generate a data.frame from the resulting SummarizedExperiment object

```{r, get_df}
df_res <- get_df_wide(dep)
```

## Save data

Save R objects:

```{r, save_load, eval = T}
save(data_diff, data_imp, data_norm, data_se, dep, file = 'intermediate/proteomic_FAK_adipose_noIP1.RData')
# load('intermediate/proteomic_FAK_adipose.RData')
```

Write results to a table:

```{r, eval = F}
write.table(df_res,
            file = 'results/table/proteomic_FAK_adipose_results_noIP1.tsv',
            quote = T,
            row.names = F,
            sep = '\t')
```

## Plots for manuscript

Function to make a publication plot:

```{r}
plot_volcano_publication <- function(dep, contrast, label_size = 3, add_names = TRUE, adjusted = FALSE, 
                         plot = TRUE, down_side, up_side, lable_genes) {
    if (is.integer(label_size)) 
        label_size <- as.numeric(label_size)
    assertthat::assert_that(inherits(dep, "SummarizedExperiment"), 
                            is.character(contrast), length(contrast) == 1, is.numeric(label_size), 
                            length(label_size) == 1, is.logical(add_names), length(add_names) == 
                                1, is.logical(adjusted), length(adjusted) == 1, is.logical(plot), 
                            length(plot) == 1)
    row_data <- SummarizedExperiment::rowData(dep, use.names = FALSE)
    if (any(!c("name", "ID") %in% colnames(row_data))) {
        stop(paste0("'name' and/or 'ID' columns are not present in '", 
                    deparse(substitute(dep)), "'.\nRun make_unique() to obtain required columns."), 
             call. = FALSE)
    }
    if (length(grep("_p.adj|_diff", colnames(row_data))) < 1) {
        stop(paste0("'[contrast]_diff' and '[contrast]_p.adj' columns are not present in '", 
                    deparse(substitute(dep)), "'.\nRun test_diff() to obtain the required columns."), 
             call. = FALSE)
    }
    if (length(grep("_significant", colnames(row_data))) < 1) {
        stop(paste0("'[contrast]_significant' columns are not present in '", 
                    deparse(substitute(dep)), "'.\nRun add_rejections() to obtain the required columns."), 
             call. = FALSE)
    }
    if (length(grep(paste(contrast, "_diff", sep = ""), colnames(row_data))) == 
        0) {
        valid_cntrsts <- row_data %>% data.frame() %>% select(ends_with("_diff")) %>% 
            colnames(.) %>% gsub("_diff", "", .)
        valid_cntrsts_msg <- paste0("Valid contrasts are: '", 
                                    paste0(valid_cntrsts, collapse = "', '"), "'")
        stop("Not a valid contrast, please run `plot_volcano()` with a valid contrast as argument\n", 
             valid_cntrsts_msg, call. = FALSE)
    }
    diff <- grep(paste(contrast, "_diff", sep = ""), colnames(row_data))
    if (adjusted) {
        p_values <- grep(paste(contrast, "_p.adj", sep = ""), 
                         colnames(row_data))
    }
    else {
        p_values <- grep(paste(contrast, "_p.val", sep = ""), 
                         colnames(row_data))
    }
    signif <- grep(paste(contrast, "_significant", sep = ""), 
                   colnames(row_data))
    df <- data.frame(x = row_data[, diff], y = -log10(row_data[, 
                                                               p_values]), significant = row_data[, signif], name = row_data$name) %>% 
        filter(!is.na(significant)) %>% arrange(significant)
    name1 <- gsub("_vs_.*", "", contrast)
    name2 <- gsub(".*_vs_", "", contrast)
    p <- ggplot(df, aes(x, y)) + 
        geom_point(aes(col = significant), size = 2.5, alpha = 0.7) + 
        geom_text(data = data.frame(), 
                  aes(x = c(Inf, -Inf), y = c(-Inf, -Inf), hjust = c(1.1, -0.1), vjust = c(-1, -1), label = c(up_side, down_side), size = 5, 
                      fontface = "italic")) + 
        labs(x = expression(bold(log[2] ~ "Fold change")), y = if (adjusted) expression(bold(-log[10] ~ "Adjusted p-value")) else expression(bold(-log[10] ~ "P-value"))) +
        geom_vline(xintercept = 0, col='darkgrey') +
        theme_minimal() + 
        theme(legend.position = "none",
              #panel.grid = element_blank(),  # Remove grid lines
              axis.text = element_text(size = 12, color = "black", face = "bold"),  # Bold axis text
              axis.title = element_text(size = 14, color = "black", face = "bold"),  # Bold axis titles
              plot.title = element_text(size = 16, face = "bold", hjust = 0.5, color = "black"),
              plot.caption = element_text(size = 10, color = "black"),
              axis.line = element_line(size = 0.5),  # Adjust axis line thickness
              axis.line.x = element_line(color = "black", size = 0.5),  # X-axis line settings
              axis.line.y = element_line(color = "black", size = 0.5),  # Y-axis line settings
              axis.line.top = element_blank(),  # Remove top axis line
              axis.line.right = element_blank(), # Remove right axis line
              axis.ticks = element_line(color = "black")) +  # Add axis ticks
      scale_color_manual(values = c(`TRUE` = "black", `FALSE` = "grey"))
    if (add_names) {
        p <- p + ggrepel::geom_text_repel(data = filter(df, significant),
          # p + ggrepel::geom_text_repel(data = filter(df, name %in% lable_genes),
                                          aes(label = name), size = label_size, box.padding = unit(0.1,
                                                                                                   "lines"), point.padding = unit(0.1, "lines"),
                                          segment.size = 0.5,
                                          nudge_y = 0.05,  # Adjust this value to move labels up
                                          force = 1,  # Adjust force to control repulsion
                                          vjust = -0.5,  # Vertical justification to center labels on dots
                                          color = "black")  # Ensure labels are black
    }
    if (plot) {
        return(p)
    }
    else {
        df <- df %>% select(name, x, y, significant) %>% arrange(desc(x))
        colnames(df)[c(1, 2, 3)] <- c("protein", "log2_fold_change", 
                                      "p_value_-log10")
        if (adjusted) {
            colnames(df)[3] <- "adjusted_p_value_-log10"
        }
        return(df)
    }
}
```

FAK_PF_INS vs FAK_DMSO_INS:

```{r, fig3B, fig.height = 5, fig.width = 5}
# Example usage
fig3B <- plot_volcano_publication(dep, contrast = "FAK_DMSO_INS_vs_FAK_DMSO_basal",
                                  down_side = "DMSO",
                                  up_side = "DMSO + insulin")
fig3B

pdf('results/figures/fig3B_FAK_DMSO_INS_vs_FAK_DMSO_basal_guide_lines.pdf',  height = 5, width = 5)
  print(fig3B)
dev.off()
```

FAK_PF_INS vs FAK_DMSO_INS:

```{r, fig3C, fig.height = 5, fig.width = 5}
# Example usage
fig3C <- plot_volcano_publication(dep, contrast = "FAK_PF_INS_vs_FAK_DMSO_INS",
                                  down_side = "DMSO + insulin",
                                  up_side = "PF + insulin"
                                  #lable_genes = FAK_PF_INS_vs_FAK_DMSO_INS_sign_names
                                  )
fig3C

pdf('results/figures/fig3C_FAK_PF_INS_vs_FAK_DMSO_INS_guide_lines.pdf',  height = 5, width = 5)
  print(fig3C)
dev.off()
```

## Session information

```{r, session_info, echo = FALSE}
sessionInfo()
```
