---
title: " Mass-spec proteomics analysis: Focal adhesion kinase (FAK) in adipose cells (no IP1)"
author:
  name: "Dmytro Kryvokhyzha"
email: dmytro.kryvokhyzha@med.lu.se
affiliation: LUDC Bioinformatics Unit
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: true
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE, warning=FALSE, eval=TRUE)
knitr::opts_knit$set(root.dir = '~/Projects/Karin_Stenkula/proteomic_FAK_adipose/')
```

## Dependencies

```{r, message=FALSE, eval=TRUE}
library(readxl)
library(DEP)
library(dplyr)
```

## Data

### Load

Load the data from the Excel file:

```{r}
data <- read_excel("data/Karin_Stenkula_IP_202312_LH_filtered.xlsx")
```

### Process

Check the structure of the data:

```{r}
str(data)
```

Are there any duplicated accession names?

```{r}
data$v %>% duplicated() %>% any()
```

## Remove IP1 

I remove the sample `IP1` because it is an outlier in a PCA plot
of the previous analysis with all samples. See `results/reports/proteomic_FAK_adipose.html`.

```{r}
data_no_IP1 <- data[,-grep('IP1,', colnames(data))]
data <- data_no_IP1
```

## Experimental design

Create an experimental design table:

```{r}
colnames(data)[c(11:21)] <- gsub(' ', '_', gsub(', ', '_', colnames(data)[c(11:21)]))
labels <- colnames(data)[c(11:21)]

expDesign <- data.frame(label = labels,
                        condition = sub("^[^_]+_", "", labels),
                        replicate = sub("_.+", "", labels)
                        )
knitr::kable(expDesign)
```

Generate a SummarizedExperiment object:

```{r}
# create the ID and gene name columns:
colnames(data)[2] <- 'ID'
data$name <- sub("^.*\\|([^|]+)_.*$", "\\1", data$`FASTA Title Lines`)

# Generate a SummarizedExperiment object
data_se <- make_se(data, c(11:21), expDesign)
data_se
```

## Filter

Plot a barplot of the protein identification overlap between samples

```{r, fig.width = 4, fig.height = 4}
plot_frequency(data_se)
```

Filter for proteins that are identified in all replicates of at least one condition

```{r}
data_filt <- filter_missval(data_se, thr = 0)
# Less stringent filtering:
# data_filt2 <- filter_missval(data_se, thr = 1)
```

Plot a barplot of the number of identified proteins per samples

```{r, fig.width = 5, fig.height = 6}
plot_numbers(data_filt)
```

Plot a barplot of the protein identification overlap between samples

```{r, fig.width = 4, fig.height = 4}
plot_coverage(data_filt)
```

## Normalize

Normalize the data

```{r}
data_norm <- normalize_vsn(data_filt)
```

Verify the fit:

```{r, fig.width = 5, fig.height = 4}
meanSdPlot(data_norm)
```

Visualize normalization by boxplots for all samples before and after normalization

```{r, fig.width = 6, fig.height = 6}
plot_normalization(data_filt, data_norm)
```

## Impute data for missing values

Plot a heatmap of proteins with missing values

```{r, fig.height = 6, fig.width = 5}
plot_missval(data_filt)
```

Plot intensity distributions and cumulative fraction of proteins with and without missing values

```{r, plot_detect, fig.height = 4, fig.width = 4}
plot_detect(data_filt)
```

The data can be missing at random (MAR), for example if proteins are quantified 
in some replicates but not in others. Data can also be missing not at random (MNAR), 
for example if proteins are not quantified in specific conditions (e.g. in the control samples).

MNAR should be imputed by a left-censored imputation method, such as the quantile 
regression-based left-censored function (“QRILC”) or random draws from a left-shifted 
distribution (“MinProb” and “man”). In contrast, MAR data should be imputed with 
methods such as k-nearest neighbor (“knn”) or maximum likelihood (“MLE”) functions. 

Impute:

```{r, impute, results = "hide", message = FALSE, warning = FALSE, error = TRUE}
data_imp <- impute(data_norm, fun = "knn")
```

Plot intensity distributions before and after imputation

```{r, plot_imp, fig.width = 5, fig.height = 4}
plot_imputation(data_norm, data_imp)
```

## Differential enrichment analysis

Based on linear models and empirical Bayes statistics

```{r, statistics}
data_diff <- test_diff(data_imp, type = "all")
```

Denote significant proteins based on user defined cutoffs

```{r, add_reject}
dep <- add_rejections(data_diff, alpha = 0.05, lfc = 0.5)
```

## Visualization of the results

### PCA

Plot the first and second principal components

```{r, pca, fig.height = 3, fig.width = 6}
plot_pca(dep, x = 1, y = 2, point_size = 4, indicate = "condition")
```

### Correlation matrix

Plot the Pearson correlation matrix

```{r, corr, fig.height = 6, fig.width = 7}
plot_cor(dep, indicate = "condition")
```

### Heatmap

Plot a heatmap of all significant proteins with the data centered per protein

```{r, heatmap, fig.height = 6, fig.width = 7}
plot_heatmap(dep,
             type = "centered",
             kmeans = TRUE,
             show_row_names = FALSE,
             indicate = "condition")
```

Plot a heatmap of all significant proteins (rows) and the tested contrasts (columns)

```{r, heatmap2, fig.height = 8, fig.width = 4}
plot_heatmap(dep, type = "contrast", kmeans = TRUE, 
             k = 6, col_limit = 10, show_row_names = FALSE)
```

### Volcano plot

Plot a volcano plot for the contrast FAK contrasts:

```{r, volcano, fig.height = 5, fig.width = 5}
plot_volcano(dep, contrast = "FAK_DMSO_basal_vs_FAK_DMSO_INS")
plot_volcano(dep, contrast = "FAK_DMSO_basal_vs_FAK_PF_INS")
plot_volcano(dep, contrast = "FAK_DMSO_INS_vs_FAK_PF_INS")
```

### Barplot

Plot a barplot for top 4 proteins from each comparison:

```{r, bar, fig.height = 8, fig.width = 5}
plot_single(dep,
            proteins = c('CAH4', 'GNAI2', 'ACAD9', 'ARPC4', 'DCMC', 'ARPC2'),
            type = "contrast")
```

Plot a barplot for the same protein with the centered log2-intensity data:

```{r, bar2, fig.height = 8, fig.width = 5}
plot_single(dep,
            proteins = c('CAH4', 'GNAI2', 'ACAD9', 'ARPC4', 'DCMC', 'ARPC2'),
            type = "centered")
```

### Frequency plots

Plot a frequency plot of significant proteins for the different contrasts:

```{r, overlap, fig.height = 5, fig.width = 8}
plot_cond(dep)
```
Print the overlap frequency table:

```{r, overlap+_print}
dep_overlap <- plot_cond(dep, plot = F)
knitr::kable(cbind(dep_overlap$counts))
knitr::kable(dep_overlap$legend)
```

## Results table

Generate a results table

```{r, results_table}
data_results <- get_results(dep)
```

Number of significant proteins

```{r, results_table_sign}
data_results %>% filter(significant) %>% nrow()
```

Generate a data.frame from the resulting SummarizedExperiment object

```{r, get_df}
df_res <- get_df_wide(dep)
```

## Save data

Save R objects:

```{r, save_load, eval = FALSE}
save(data_diff, data_imp, data_norm, data_se, dep, file = 'intermediate/proteomic_FAK_adipose_noIP1.RData')
# load('intermediate/proteomic_FAK_adipose.RData')
```

Write results to a table:

```{r}
write.table(df_res,
            file = 'results/table/proteomic_FAK_adipose_results_noIP1.tsv',
            quote = T,
            row.names = F,
            sep = '\t')
```


## Session information

```{r, session_info, echo = FALSE}
sessionInfo()
```
